week-1/implement/dedupe.js

```diff
@@ -1 +1,7 @@
-function dedupe() {}
+function dedupe(arr) {
+    if(arr.length === 0){
```

Comment:
Do you need this special check for an empty array? Or the `new Set` command will handle it without our help?

Reply:
No without this check the code works I only use this check if Array is empty the function returns empty array without proceeding other code

Reply:
I see. The motivation is good. 
But no need really. Programmers often call it "premature optimisation".
There should be good reasons to add complexity to the code.
If your code works without the optimisation, then you don't need that.